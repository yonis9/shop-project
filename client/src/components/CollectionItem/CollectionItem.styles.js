import styled from 'styled-components';
import CustomButton from '../CustomButton/CustomButton';

export const CollectionItemContainer = styled.div`
    width: 22vw;
    display: flex;
    flex-direction: column;
    height: 350px;
    align-items: center;
    position: relative;

    &:hover {
        .image {
            opacity: 0.8;
          }

        button {
          display: flex;
          opacity: 0.9;
        }
`

export const AddButton = styled(CustomButton)`
  width: 80%;
  opacity: 0.7;
  position: absolute;
  top: 255px;
  display: none;
`;

export const BackgroundImage = styled.div`
    width: 100%;
    height: 95%;
    background-size: cover;
    background-position: center;
    margin-bottom: 5px;
    background-image: url(${(props) => props.imageUrl})
`
export const CollectionFooterContainer = styled.div`
    padding:  0 20px;
    width: 100%;
    height: 5%;
    display: flex;
    justify-content: space-evenly;
    font-size: 18px;
`

export const NameContainer = styled.span`
    width: 90%;
    margin-bottom: 15px;
`

export const PriceContainer = styled.span`
    width: 10%;
    text-align: right;
`



