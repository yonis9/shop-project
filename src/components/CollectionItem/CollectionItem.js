import React from 'react';

import CustomButton from '../CustomButton/CustomButton';

import './CollectionItem.scss';

const CollectionItem = ( { name, imageUrl, price }) => (
    <div className='collection-item'>
        <div className='image'
             style={{backgroundImage: `url(${imageUrl})`}}/>
        <div className='collection-footer'>
            <span className='name'>{name}</span>
            <span className='price'>${price}</span>
        </div>
        <CustomButton inverted>ADD TO CART</CustomButton>
    </div>
)

export default CollectionItem