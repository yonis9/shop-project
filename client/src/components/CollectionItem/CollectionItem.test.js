import React from 'react';
import { shallow } from 'enzyme';

import { CollectionItem } from './CollectionItem';


describe('CollectionItem component', () => {

    let wrapper;
    let moackAddItem;
    const mockName = 'shirt';
    const mockUrl = 'shirts.com';
    const mockPrice = 15;

    const mockItem = {
        name: mockName,
        imageUrl: mockUrl,
        price: mockPrice
    }

    beforeEach(() => {
        moackAddItem = jest.fn();
        wrapper = shallow(<CollectionItem item={mockItem} addItem={moackAddItem}/>)
    })

    it('should rendered CollectionItem component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should called addItem with Item', () => {
        wrapper.find('AddButton').simulate('click');
        expect(moackAddItem).toHaveBeenCalledWith(mockItem)
    });

    it('should render imageUrl as a prop on BackgroundImage', () => {
        expect(wrapper.find('BackgroundImage').prop('imageUrl')).toBe(mockUrl);
      });
    
    it('should render name in NameContainer', () => {
        expect(wrapper.find('NameContainer').text()).toBe(mockName);
    });

    it('should render price in PriceContainer', () => {
        expect(wrapper.find('PriceContainer').text()).toBe(`$${mockPrice}`);
    });

});