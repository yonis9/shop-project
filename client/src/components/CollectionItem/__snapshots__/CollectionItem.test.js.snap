// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionItem component should rendered CollectionItem component 1`] = `ShallowWrapper {}`;
